plugins/linguist-backend/src/api/LinguistBackendApi.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  EntitiesOverview,
+  EntityResults,
+  Language,
+  Languages,
+} from '@backstage/plugin-linguist-common';
+import {
+  CATALOG_FILTER_EXISTS,
+  CatalogClient,
+  GetEntitiesRequest,
+} from '@backstage/catalog-client';
+import { PluginEndpointDiscovery, UrlReader } from '@backstage/backend-common';
+
+import { Config } from '@backstage/config';
+import { DateTime } from 'luxon';
+import { LINGUIST_ANNOTATION } from '@backstage/plugin-linguist-common';
+import { LinguistBackendStore } from '../db';
+import { Logger } from 'winston';
+import fs from 'fs-extra';
+import linguist from 'linguist-js';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { assertError } from '@backstage/errors';
+
+/** @public */
+export class LinguistBackendApi {
+  public constructor(
+    private readonly config: Config,
+    private readonly logger: Logger,
+    private readonly store: LinguistBackendStore,
+    private readonly urlReader: UrlReader,
+    private readonly discovery: PluginEndpointDiscovery,
+  ) {}
+
+  public async getEntityLanguages(entity: string): Promise<Languages> {
+    this.logger?.debug(`Getting languages for entity "${entity}"`);
+
+    return this.store.getEntityResults(entity);
+  }
+
+  public async processEntity(entity: string, url: string): Promise<string> {
+    this.logger?.info(`Processing languages for entity ${entity} from ${url}`);
+
+    const readTreeResponse = await this.urlReader.readTree(url);
+    const dir = await readTreeResponse.dir();
```

Comment:
Yes, that's something I've considered. From what I remember both GitHub and GitLab provide this data via their API. For now I'd like to put this on the To Do list if you don't mind so we can get the plugin in a good state and build on it. I don't mind adding more features as it a lot easier because the PRs end up being a lot smaller.

Reply:
Alright!